op-verifier/contracts/OPDisputeGameVerifier.sol

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+import { IEVMVerifier } from "@ensdomains/evm-verifier/contracts/IEVMVerifier.sol";
+import { RLPReader } from "@eth-optimism/contracts-bedrock/src/libraries/rlp/RLPReader.sol";
+import { StateProof, EVMProofHelper } from "@ensdomains/evm-verifier/contracts/EVMProofHelper.sol";
+import { Types } from "@eth-optimism/contracts-bedrock/src/libraries/Types.sol";
+import { Hashing } from "@eth-optimism/contracts-bedrock/src/libraries/Hashing.sol";
+import "./interfaces/IDisputeGame.sol";
+import "./interfaces/IDisputeGameFactory.sol";
+
+struct OPWitnessData {
+    uint256 disputeGameIndex;
+    Types.OutputRootProof outputRootProof;
+}
+
+// https://docs.optimism.io/builders/notices/fp-changes#overview-of-changes
+contract OPDisputeGameVerifier is IEVMVerifier {
+    error OutputRootMismatch(uint256 disputeGameIndex, bytes32 expected, bytes32 actual);
+    error GameTypeMismatch(uint256 disputeGameIndex, GameType expected, GameType actual);
+
+    IDisputeGameFactory public disputeGameFactory;
+    string[] _gatewayURLs;
+
+    // The game type that the OptimismPortal consults for output proposals.
+    GameType public respectedGameType;
+
+    constructor(string[] memory urls, address game, GameType gameType) {
```

Comment:
This should take an argument for the minimum age of the proof (in blocks / epochs); otherwise, it's entirely up to the gateway and it could choose to return a just-submitted dispute game that's had no chance to be challenged yet.

Reply:
I have added delay check in timestamp to the OPDisputeGameVerifier but the original OPVerifier doesn't seem to have a delay check implemented on the contract.